algorithms/src/Kokkos_RadixSort.hpp

```diff
@@ -0,0 +1,363 @@
+  KOKKOS_FUNCTION
+  KeyFromView(KeyView const& k, std::integral_constant<int, BitWidth>)
+      : keys(k) {}
+
+  // i: index of the key to get
+  // bit: which bit, with 0 indicating the least-significant
+  template <typename IndexType>
+  KOKKOS_INLINE_FUNCTION auto operator()(IndexType i, std::uint32_t bit) const {
+    // I'd rather use a fully aliasing view here ...
+    auto key = *reinterpret_cast<key_integral_type*>(&keys(i));
+    auto h   = key >> bit;
+
+    // Handle the sign bit of signed 2's-complement indicating low values
+    if constexpr (std::is_integral_v<key_value_type> &&
+                  std::is_signed_v<key_value_type>) {
+      if (bit == 8 * sizeof(key_value_type) - 1) {
+        return h & 0b1u;
+      }
+    }
+
+    // Handle the sign-magnitude representation of FP
+    if constexpr (std::is_floating_point_v<key_value_type>) {
+      auto sign_bit_pos = 8 * sizeof(key_value_type) - 1;
+      auto sign_bit     = key >> sign_bit_pos;
+      if (bit == sign_bit_pos) {
+        return h & 0b1u;
+      } else {
+        return ~(h ^ sign_bit) & 0b1u;
+      }
+    }
+
+    return ~h & 0b1u;
+  }
+
+  KOKKOS_FUNCTION
+  int getNumBits() { return num_bits; }
+
+  KOKKOS_FUNCTION
+  size_t extent() { return keys.extent(0); }
+};
+
+template <typename KeyView>
+KeyFromView(KeyView const&)
+    ->KeyFromView<8 * sizeof(typename KeyView::value_type), KeyView>;
+
+template <typename KeyView, int BitWidth>
+KeyFromView(KeyView const&, std::integral_constant<int, BitWidth>)
+    ->KeyFromView<BitWidth, KeyView>;
+
+template <typename KeyType,
+          typename MemorySpace = Kokkos::DefaultExecutionSpace::memory_space,
+          typename IndexType   = ::std::uint32_t>
+class RadixSorter {
+ public:
+  RadixSorter() = default;
+
+  template <typename ExecutionSpace>
+  explicit RadixSorter(ExecutionSpace const& exec, std::size_t n)
+      : m_key_scratch(
+            Kokkos::view_alloc(exec, Kokkos::WithoutInitializing,
+                               "Kokkos::RadixSorter radix_sort_key_scratch"),
+            n),
+        m_index_old(Kokkos::view_alloc(exec, Kokkos::WithoutInitializing,
+                                       "Kokkos::RadixSorter radix_sort_index"),
+                    n),
+        m_index_new(
+            Kokkos::view_alloc(exec, Kokkos::WithoutInitializing,
+                               "Kokkos::RadixSorter radix_sort_index_scratch"),
+            n),
+        m_scan(Kokkos::view_alloc(exec, Kokkos::WithoutInitializing,
+                                  "Kokkos::RadixSorter radix_sort_scan"),
+               n),
+        m_bits(Kokkos::view_alloc(exec, Kokkos::WithoutInitializing,
+                                  "Kokkos::RadixSorter radix_sort_bits"),
+               n) {}
+
+  // Generate and store the permutation induced by the keys, without
+  // modifying their initial order
+  template <typename ExecutionSpace, typename KeyFunctor>
+  void create_indirection_vector(ExecutionSpace const& exec,
```

Comment:
If I read the comment you wrote about what this method is supposed to do I kind of understand what it does, but the name `create_indirection_vector` is a bit too generic and does not reflect what it is doing. I would suggest something like: `generate_and_store_permutation_vector` or something similar. Also because `create_` here does not return anything but stores it inside